### ISSUE TITLE
Djangoforms should be removed

### ISSUE THREAD

**Issue by sylvainvivien** [Arachnid/bloggart#24]
I think Djangoforms should be removed for 2 reasons :)

1 - Personaly, (so not very important :p ), I think that djangoforms is overkill for only 3 text fields.

2 - Most important this time : Djangoforms goes against the second rule of bloggart "Good isolation of code and markup. Users shouldn't have to understand the innards of our blogging software if all they want to do is change how the blog looks or is laid out." and particulalry for 2 things :
- theming : for example height, width,... are set inside .py files and depend on the Djangoform. Not very easy to change this if you want to skin this part.
- i18n : I'd like to support english and french for the admin part, but currently I can't translate the labels "title", "body" and "tags". They are part of the djangoforms.

So I think it could be a nice feature to remove Djangoforms and to use for example jquery to check the forms (mandatory fields) and add a nice alert box (jalert : http://abeautifulsite.net/notebook_files/87/demo/)... and maybe start a little Ajax (for the preview for example,...)

And example of jquery code to validate the form (it has to be checked server side too)

 <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.min.js" type="text/javascript"></script>
  <script type="text/javascript">

```
$(function() {
  $("input[name=title]").select();
  $("form.compose").submit(function() {
      var required = ["title", "body", "tags"];
      var form = $(this).get(0);
      for (var i = 0; i < required.length; i++) {
          if (!form[required[i]].value) {
              $(form[required[i]]).select();
              return false;
          }
      }
      return true;
  });
});
```

**Comment by Arachnid** [comment]
Good point about form generation and templating - I concur. I'm not so sure about giving up on it for _validation_, though.

**Comment by Arachnid** [comment]
After having a go at rewriting the server-side logic first to not use Djangoforms at all, then to use it for validation only (with standard forms instead of modelforms), I've concluded that doing so is more effort than it's worth - I think that even in this case, use of a forms library brings more clarity than it loses by hiding the markup.

**Comment by sylvainvivien** [comment]
I really think that skin, template,... easy i18n or being abble to change any labels, size, width, font, easily (only HTML).... is more important than speed, server side validation,.... and particularly if you want Bloggart to be adopted by a lot of users (non english and with no python dev skill), this is the target I think.

So, IMHO, djangoforms can be a brake.

**Comment by Arachnid** [comment]
I agree with your points - but implementing the editing process without a forms library remains onerous. If you want to submit a patch that removes Django forms and replicates the functionality, I'll accept it. :)
